GitTfs/Core/GitTfsRemote.cs

```diff
@@ -325,9 +325,12 @@ public IFetchResult FetchWithMerge(int mergeChangesetId, bool stopOnFailMergeCom
             var latestChangesetId = GetLatestChangesetId();
             if (lastChangesetIdToFetch != -1)
                 latestChangesetId = Math.Min(latestChangesetId, lastChangesetIdToFetch);
-            // TFS 2010 doesn't like when we ask for history past its last changeset.
+
             if (MaxChangesetId >= latestChangesetId)
+            {
+                Trace.WriteLine("TFS 2010 doesn't like when we ask for history past its last changeset.");
```

Comment:
I guess "a weird state" is a good way to describe my current situation. After several hours of debugging I'm still not entirely sure what is wrong... But anyway I think logging the message could have been of some help instead of simply failing silently.